target_logic4/sinks.py

```diff
@@ -29,12 +29,9 @@ def preprocess_record(self, record: dict, context: dict) -> dict:
                 "ProductId": int(product_id) if product_id else None,
                 "QtyToOrder": item.get("quantity"),
                 "QtyToDeliver": item.get("quantity"),
-                "OrderedOnDateByDistributor": record.get("transaction_date")
+                "OrderedOnDateByDistributor":  created_at,
+                "ExpectedDeliveryDate": record.get("created_at")
             }
-            if item.get("receipt_date"):
-                line_item["ExpectedDeliveryDate"] = item.get("receipt_date")
-
-            PurchaseOrderLines.append(line_item)
```

Comment:
This line shouldn't be removed

Reply:
I added that directly to payload.

If want to leave the extra verification, has to be like this because value comes from created_at:
`
            if item.get("created_at"):
                line_item["ExpectedDeliveryDate"] = item.get("created_at")

            PurchaseOrderLines.append(line_item)
`

Reply:
I agree the mapping for `ExpectedDeliveryDate` is fine, what I mean is the line below that should be kept or we won't be adding the lines to the payload:

`PurchaseOrderLines.append(line_item)`